fastai_plugin/semantic_segmentation_backend.py

```diff
@@ -154,6 +155,42 @@ def _write_zip(results, split):
 
         upload_or_copy(group_path, group_uri)
 
+    def subset_training_data(self, chip_dir):
+        """ Specify a subset of all the training chips that have been created
+
+        This creates uses the train_opts 'train_proportion' parameter to
+            subset a number (n) of the training chips. It creates two new 
+            directories 'train-{n}-img' and 'train-{n}-labels' with
+            subsets of the chips that the dataloader can read from.
+
+        Args:
+            chip_dir (str): path to the chip directory
+
+        Returns:
+            (str) name of the train subset image directory (e.g. 'train-{n}-img')
+        """        
+        random.seed(100)
+        prop = self.train_opts.train_proportion
+
+        all_train_uri = join(chip_dir, 'train-img')
+        all_train = os.listdir(all_train_uri)
+        sample_size = round(prop * len(all_train))
```

Comment:
I would also sort `all_train` before doing the sampling to make things repeatable. This is because `os.listdir` returns things in an arbitrary order.